### ISSUE TITLE
Error: EMFILE, too many open files '/home/ubuntu/***/views/auth'

### ISSUE THREAD

**Issue by RSSGergelyMunkacsy** [balderdashy/sails#2741]
I get the following error if too many user visit my sever:
```
[36mverbose [39m: Shutting down socket server...
Error:
  code: 'EMFILE',
  path: '/home/ubuntu/***/views/auth',
  syscall: 'readdir' }
 [31merror [39m: Sending 500 ("Server Error") response:
 Error: EMFILE, too many open files '/home/ubuntu/***/views/auth'
    at Object.fs.readdirSync (fs.js:665:18)
    at Glob._readdir (/home/ubuntu/***/node_modules/sails/node_modules/glob/glob.js:662:20)
    at Glob._process (/home/ubuntu/***/node_modules/sails/node_modules/glob/glob.js:446:15)
    at Glob.iterator (/home/ubuntu/***/node_modules/sails/node_modules/glob/glob.js:181:10)
    at Array.forEach (native)
    at new Glob (/home/ubuntu/***/node_modules/sails/node_modules/glob/glob.js:179:22)
    at glob (/home/ubuntu/***/node_modules/sails/node_modules/glob/glob.js:57:11)
    at Function.globSync [as sync] (/home/ubuntu/***/node_modules/sails/node_modules/glob/glob.js:76:10)
    at globPath (/home/ubuntu/***/node_modules/sails/lib/hooks/http/view.js:17:15)
    at exists (/home/ubuntu/***/node_modules/sails/lib/hooks/http/view.js:23:10)
    at SailsView.lookup (/home/ubuntu/***/node_modules/sails/lib/hooks/http/view.js:40:6)
    at View (/home/ubuntu/***/node_modules/sails/node_modules/express/lib/view.js:44:20)
    at new SailsView (/home/ubuntu/***/node_modules/sails/lib/hooks/http/view.js:32:15) { [Error: EMFILE, too many open files '/home/ubuntu/***/views/auth']
  errno: 20,
  code: 'EMFILE',
  path: '/home/ubuntu/***/views/auth',
  syscall: 'readdir' }
```

My Sails version: `0.10.5`
I increase the `ulimit -n` from 1024 to 2048 but I think this is not a real solution.

**Comment by crh3675** [comment]
@RSSGergelyMunkacsy Perhaps too many users is a good problem to have.  You aren't alone in the scalability woes of having to update ulimit settings.  That actually is the first thing I do when I setup new VMs because I am typically building apps that expect over 1k users hitting them at one time.  I don't consider it to be a problem as it is naturally part of my solution to set that number about 64k.

**Comment by CWyrtzen** [comment]
So, is this an issue/error or did @crh3675 clear things up? Can we close this?

**Comment by sailsbot** [comment]
Thanks for posting, @RSSGergelyMunkacsy.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
